sklearn/impute/_base.py

```diff
@@ -213,16 +218,23 @@ class SimpleImputer(_BaseImputer):
     """
     @_deprecate_positional_args
     def __init__(self, *, missing_values=np.nan, strategy="mean",
-                 fill_value=None, verbose=0, copy=True, add_indicator=False):
+                 fill_value=None, verbose=None, copy=True,
+                 add_indicator=False):
         super().__init__(
             missing_values=missing_values,
             add_indicator=add_indicator
         )
         self.strategy = strategy
         self.fill_value = fill_value
-        self.verbose = verbose
         self.copy = copy
 
+        if verbose is not None:
+            warnings.warn("The 'verbose' parameter was deprecated in version "
+                          "1.0 and will be removed in 1.2. A warning will "
+                          "always be raised upon the removal of empty columns "
+                          "in the future version.", FutureWarning)
```

Comment:
We do input validation in `fit`. You can find out how to do this in the [development user guide](https://scikit-learn.org/stable/developers/contributing.html#deprecation).